src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IPatternSwitchCase.cs

```diff
@@ -0,0 +1,679 @@
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS0266: Cannot implicitly convert type 'object' to 'bool'. An explicit conversion exists (are you missing a cast?)
+                //             /*<bind>*/case X y when x:/*</bind>*/
+                Diagnostic(ErrorCode.ERR_NoImplicitConvCast, "x").WithArguments("object", "bool").WithLocation(9, 37)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<CasePatternSwitchLabelSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact, WorkItem(19927, "https://github.com/dotnet/roslyn/issues/19927")]
+        public void TestPatternCaseClause_ConstantPattern()
+        {
+            string source = @"
+using System;
+class X
+{
+    void M(bool x)
+    {
+        switch (x)
+        {
+            case /*<bind>*/x is true/*</bind>*/:
+                break;
+        }
+    }
+}
+";
+            string expectedOperationTree = @"
+IIsPatternExpression (OperationKind.IsPatternExpression, Type: System.Boolean) (Syntax: 'x is true')
+  Expression: IParameterReferenceExpression: x (OperationKind.ParameterReferenceExpression, Type: System.Boolean) (Syntax: 'x')
+  Pattern: IConstantPattern (OperationKind.ConstantPattern) (Syntax: 'true')
+      Value: ILiteralExpression (OperationKind.LiteralExpression, Type: System.Boolean, Constant: True) (Syntax: 'true')
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS0150: A constant value is expected
+                //             case /*<bind>*/x is true/*</bind>*/:
+                Diagnostic(ErrorCode.ERR_ConstantExpected, "x is true").WithLocation(9, 28)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<IsPatternExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact, WorkItem(19927, "https://github.com/dotnet/roslyn/issues/19927")]
+        public void TestPatternCaseClause_DefaultLabel()
+        {
+            string source = @"
+using System;
+class X
+{
+    void M(object x)
+    {
+        switch (x)
+        {
+            case X y:
+            /*<bind>*/default:/*</bind>*/
+                break;
+        }
+    }
+}
+";
+            string expectedOperationTree = @"
+ISwitchStatement (1 cases) (OperationKind.SwitchStatement) (Syntax: 'switch (x) ... }')
+  Switch expression: IParameterReferenceExpression: x (OperationKind.ParameterReferenceExpression, Type: System.Object) (Syntax: 'x')
+  Sections: ISwitchCase (2 case clauses, 1 statements) (OperationKind.SwitchCase) (Syntax: 'case X y: ... break;')
+        Clauses: IPatternCaseClause (Label Symbol: case X y:) (OperationKind.PatternCaseClause) (Syntax: 'case X y:')
+            Pattern: IDeclarationPattern (Declared Symbol: X y) (OperationKind.DeclarationPattern) (Syntax: 'X y')
+          IPatternCaseClause (Label Symbol: default:) (OperationKind.PatternCaseClause) (Syntax: 'default:')
```

Comment:
We do not have a dedicated IOperation for default label, but do have a different case kind: http://source.roslyn.io/#Microsoft.CodeAnalysis/Operations/CaseKind.cs,65f316f2e0138ff5,references. Current approach now seems wrong to me as the case kind will be Pattern for the default label, it should instead be Default, which will be the case with your original suggestion (I also need to update the dump to also print the case kind).